### PR TITLE
Enable zoom on vertical axis

### LINKED ISSUE #146: Drag-zoom vertically
Hi,

Firstly, thanks to all contributors of this project. Features brought by this plugin are amazing and I'd like to use it in one of my projects.

It's currently **not possible to zoom, in `drag` mode, on vertical axis**. Even if I set `mode: 'xy'`, zoom is applied only on the horizontal axis. I checked the source code and saw this piece of code:

```js
if (options.zoom.drag) {
  // Only want to zoom horizontal axis
  options.zoom.mode = 'x';
  // ...
}
```

Is there any reason to intentionally disallow zoom on vertical axis? It would be awesome to **allow zoom on vertical axis too**. Actually, we could zoom on each axis with a different scale factor, depending on the drag selection. If you think that would be an interesting feature, I may try to submit a PR.

> * chartjs-plugin-zoom version: 0.6.3
> * Chart.js version: 2.7.1

### PR DESCRIPTION
**Pull request by jledentu**
Hi,

This PR allows to drag-zoom vertically and make zoom on X axis independent from zoom on Y axis.
The `doZoom` now accepts two parameters to specify horizontal zoom and vertical zoom. 

Some refactoring may be possible to reduce duplicated instructions. Suggestions are welcome. 

Closes #146

### CODE REVIEW

**Comment by benmccann** [comment]
@jledentu this PR will need to be rebased

**Comment by jledentu** [comment]
It has been rebased.

**Comment by benmccann** [comment]
@jledentu sorry for the long delay in getting this PR reviewed. Can you drop the compiled file `chartjs-plugin-zoom.min.js` from the PR? The maintainers will build when cutting  a release. And would you be able to rebase `src/chart.zoom.js` against master one more time?

File: src/chart.zoom.js [hunk 1]
```diff
@@ -376,23 +385,42 @@ var zoomPlugin = {
...
+
+					if (directionEnabled(options.zoom.mode, 'y')) {
+						var offsetY = beginPoint.target.getBoundingClientRect().top;
+						startY = Math.min(beginPoint.clientY, event.clientY) - offsetY;
+						endY = Math.max(beginPoint.clientY, event.clientY) - offsetY;
+					}
+
+					var dragDistanceX = endX - startX;
+					var chartDistanceX = chartArea.right - chartArea.left;
+					var zoomX = 1 + ((chartDistanceX - dragDistanceX) / chartDistanceX );
```
Comment (benmccann) [inline src/chart.zoom.js]: Please remove the extra space before `);` (same below). After you rebase against master, you can run `gulp lint` which will automatically check the PR for style issues

File: src/chart.zoom.js [hunk 2]
```diff
@@ -287,6 +287,18 @@ function positionInChartArea(chartInstance, position) {
 		(position.y >= chartInstance.chartArea.top && position.y <= chartInstance.chartArea.bottom);
 }
 
+function getXAxis(chartInstance) {
+	var scales = chartInstance.scales;
+
+	for (var scaleId in scales) {
```
Comment (benmccann) [inline src/chart.zoom.js]: I just updated `getYAxis` in order to fix an error generated by `gulp lint`. You'll have to update this to match or else the automatic style checks will fail. https://github.com/chartjs/chartjs-plugin-zoom/commit/378ac6e040175005a7b737875f8fc414e25a657b

**Comment by benmccann** [comment]
I did the rebase and merged manually in https://github.com/chartjs/chartjs-plugin-zoom/commit/7f6fadd2cbfcc6f96283d8e8d8bb00c2a3ad692f

Thank for this great contribution! And sorry again for the delay in reviewing. Really appreciate your putting this PR together!

**Comment by PhilippMatzel** [comment]
Hey, i still running in this issue, does someone know why this happens? Im using chartjs 2.9.4 and chartjs-plugin-zoom 0.7.7.
Here is my zoom object:

      zoom: {
        zoom: {
        enabled: true,
        drag: {
            borderColor: 'rgba(225,225,225,0.9)',
            borderWidth: 10,
            backgroundColor: 'rgb(225,225,225)',
            animationDuration: 0
        },
        mode: 'xy',

        rangeMin: {
            x: null,
            y: null
        },
        rangeMax: {
            x: null,
            y: null
        },
        speed: 0.1,
        threshold: 2,
        sensitivity: 3,
        onZoom: () => {  },
        onZoomComplete: () => {  }
    }
  }
};

**Pull request closed without merging**